src/TorchSharp/Tensor/Tensor.cs

```diff
@@ -27,11 +29,27 @@ public partial class Tensor : IDisposable
             static long _totalCount = 0;
             static long _peakCount = 0;
 
+            static long _totalMemory = 0;
+            static long _peakMemory = 0;
+
+            public static List<Action<Tensor>> OnTensorCreateCallback { get; } =
+                new List<Action<Tensor>>();
+
+            public static List<Action<Tensor>> OnTensorDisposeCallback { get; } =
+                new List<Action<Tensor>>();
+
             internal Tensor(IntPtr handle)
             {
                 this.handle = handle;
                 System.Threading.Interlocked.Increment(ref _totalCount);
                 _peakCount = Math.Max(_totalCount, _peakCount);
+
+                System.Threading.Interlocked.Add(ref _totalMemory, NumberOfElements * ElementSize);
```

Comment:
This will result in overestimating the amount of memory. There are lots of ways that aliases for the same storage can be created. I actually had this in earlier, but took it out because it's not reliable.

Reply:
When doing memory benchmarks, it's useful, but it could be placed in the DisposeScope if you prefer - I realized from your comments regarding PeakCount that it was always going to be aproximate. I'll remove it for now.